homeassistant/components/kostal_plenticore/helper.py

```diff
@@ -182,9 +187,137 @@ async def _async_update_data(self) -> dict[str, dict[str, str]]:
+        name: str,
+        update_inverval: timedelta,
+        plenticore: Plenticore,
+    ) -> None:
+        """Create a new update coordinator for plenticore data."""
+        super().__init__(
+            hass=hass,
+            logger=logger,
+            name=name,
+            update_interval=update_inverval,
+        )
+        # data ids to poll
+        self._fetch = defaultdict(list)
+        self._plenticore = plenticore
+
+    def start_fetch_data(self, module_id: str, data_id: str, all_options: str) -> None:
+        """Start fetching the given data (module-id and entry-id)."""
+        self._fetch[module_id].append(data_id)
+        self._fetch[module_id].append(all_options)
+        # Force an update of all data. Multiple refresh calls
+        # are ignored by the debouncer.
+        async def force_refresh(event_time: datetime) -> None:
+            await self.async_request_refresh()
+
+        async_call_later(self.hass, 2, force_refresh)
+
+    def stop_fetch_data(self, module_id: str, data_id: str, all_options: str) -> None:
+        """Stop fetching the given data (module-id and entry-id)."""
+        self._fetch[module_id].remove(all_options)
+        self._fetch[module_id].remove(data_id)
+
+
+class SelectDataUpdateCoordinator(PlenticoreSelectUpdateCoordinator):
+    """Implementation of PlenticoreUpdateCoordinator for select data."""
+
+    async def _async_update_data(self) -> dict[str, dict[str, str]]:
+        client = self._plenticore.client
+
+        if client is None:
+            return {}
+
+        _LOGGER.debug("Fetching select %s for %s", self.name, self._fetch)
+
+        fetched_data = await self.async_get_currentoption(self._fetch)
 
         return fetched_data
 
+    async def async_get_currentoption(
+        self,
+        module_id: Union[str, Dict[str, Iterable[str]]],
+        data_id: Union[str, Iterable[str]] = None,
+    ) -> Dict[str, Dict[str, str]]:
+        """Get current option."""
+        for mid, pids in module_id.items():
+            all_options = pids[1]
+            for all_option in all_options:
+                if all_option != "None":
+                    val = await self.async_read_data(mid, all_option)
+                    for option in val.values():
+                        if option[all_option] == "1":
+                            fetched = {mid: {pids[0]: all_option}}
+                            return fetched
+
+        return {module_id: {data_id: "None"}}
+
+    async def async_read_data(self, module_id: str, data_id: str) -> [str, bool]:
+        """Writes settings back to Plenticore."""
+        client = self._plenticore.client
+
+        if client is None:
+            return False
+
+        try:
+            val = await client.get_setting_values(module_id, data_id)
+        except PlenticoreApiException:
+            return False
+        else:
+            return val
+
+    async def async_write_data(self, module_id: str, value: dict[str, str]) -> bool:
```

Comment:
Why does this need to be overridden?

Reply:
Sorry, I do not get your point. What exactly do you mean?

Reply:
I mean that the implementation of `async_write_data` in `SelectDataUpdateCoordinator` is (now was) almost exactly the same as in `SettingDataUpdateCoordinator`. Can you try to make them share one implementation instead, maybe add the read/write in a mixin class?

Reply:
added a Mixin class